addons/base_address_extended/tests/test_street_fields.py

```diff
@@ -15,44 +16,76 @@ def setUpClass(cls):
-            'name': partner_name + '-2',
-            'street_name': street_name,
-            'street_number': street_number,
-            'street_number2': street_number2,
-            'country_id': country_id,
+    def assertStreetVals(self, partner, street_data):
+        for key, val in street_data.items():
+            if isinstance(partner[key], models.BaseModel):
+                self.assertEqual(partner[key].id, val, 'Wrongly formatted street field %s: expected %s, received %s' % (key, val, partner[key]))
+            else:
+                self.assertEqual(partner[key], val, 'Wrongly formatted street field %s: expected %s, received %s' % (key, val, partner[key]))
+
+    def test_partner_create(self):
+        us_id = self.env.ref('base.us').id
+        mx_id = self.env.ref('base.mx').id
+        ch_id = self.env.ref('base.ch').id
+        input_values = [
+            {'country_id': us_id, 'street': '40/2b Chaussee de Namur'},
+            {'country_id': us_id, 'street': '40 Chaussee de Namur'},
+            {'country_id': us_id, 'street': 'Chaussee de Namur'},
+            {'country_id': mx_id, 'street': 'Av. Miguel Hidalgo y Costilla 601'},
+            {'country_id': ch_id, 'street': 'header Chaussee de Namur, 40 - 2b trailer'},
+            {'country_id': ch_id, 'street': 'header Chaussee de Namur, 40 trailer'},
+            {'country_id': ch_id, 'street': 'header Chaussee de Namur trailer'},
+        ]
+        expected = [
+            {'street_name': 'Chaussee de Namur', 'street_number': '40', 'street_number2': '2b'},
+            {'street_name': 'Chaussee de Namur', 'street_number': '40', 'street_number2': False},
+            {'street_name': 'de Namur', 'street_number': 'Chaussee', 'street_number2': False},
+            {'street_name': 'Av.', 'street_number': 'Miguel Hidalgo y Costilla 601', 'street_number2': False},
+            {'street_name': 'Chaussee de Namur', 'street_number': '40', 'street_number2': '2b'},
+            {'street_name': 'Chaussee de Namur', 'street_number': '40', 'street_number2': False},
+            {'street_name': 'Chaussee de Namur', 'street_number': False, 'street_number2': False}
+        ]
+
+        # test street -> street values (compute)
+        for partner_values, expected_vals in zip(input_values, expected):
+            partner_values['name'] = 'Test'
+            partner = self.env['res.partner'].create(partner_values)
+            self.assertStreetVals(partner, expected_vals)
+
+        # test street_values -> street (inverse)
+        for partner_values, expected_vals in zip(input_values, expected):
+            partner_values['name'] = 'Test'
+            expected_street = partner_values.pop('street')
+            partner_values.update(expected_vals)
+            partner = self.env['res.partner'].create(partner_values)
+            self.assertEqual(partner.street, expected_street)
+            self.assertStreetVals(partner, partner_values)
+
+    def test_partner_write(self):
```

Comment:
`""" Will test the compute and inverse methods of street fields when updating partner records. """`